src/library/scala/collection/immutable/ChampCommon.scala

```diff
@@ -52,8 +52,6 @@ private[immutable] abstract class Node[T <: Node[T]] {
   def sizePredicate: Int
 
   protected final def removeElement(as: Array[Int], ix: Int): Array[Int] = {
-    if (ix < 0) throw new ArrayIndexOutOfBoundsException
```

Comment:
It was originally pointed out by NthPortal or someone. (Seems like convesation is gone, since I squashed commit logs to silence CI)
The reason of redundancy is `System.arraycopy` also raises `IndexOutOfBoundsException`.

However, the goal of this PR is to add helpful message so that Scala users can easily identify bounds when error happens.

So, I think it should be 
```scala
if (ix < 0 || ix > as.length - 1) throw new ArrayIndexOutOfBoundsException("$ix is out of bounds ...")
```

- [ ] Revert and add helpful message